### PR TITLE
Add rocALUTION for ROCm 

### PR DESCRIPTION
**Pull request by srekolam**

### CODE REVIEW

File: var/spack/repos/builtin/packages/rocm-smi-lib/package.py [hunk 1]
```diff
@@ -24,7 +24,6 @@ class RocmSmiLib(CMakePackage):
 
     @run_after('install')
     def post_install(self):
-        popen('cp -R {0}/rocm_smi/lib {1}'.format(self.prefix, self.prefix))
-        popen('cp -R {0}/rocm_smi/include {1}'.format(self.prefix,
-                                                      self.prefix))
-        popen('rm -R {0}/rocm_smi'.format(self.prefix))
+        shutil.rmtree(self.prefix.lib)
+        copy_tree(self.prefix.rocm_smi,  self.prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocm-smi-lib/package.py]:
```suggestion
        install_tree(self.prefix.rocm_smi,  self.prefix)
```
The only difference is how it handle permissions.

File: var/spack/repos/builtin/packages/rocalution/package.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+       other scientific software packages."""
+
+    homepage = "https://github.com/ROCmSoftwarePlatform/rocALUTION"
+    url      = "https://github.com/ROCmSoftwarePlatform/rocALUTION/archive/rocm-3.5.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+
+    version('3.5.0', sha256='be2f78c10c100d7fd9df5dd2403a44700219c2cbabaacf2ea50a6e2241df7bfe')
+
+    depends_on('cmake@3.5.2', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocalution/package.py:25]:
```suggestion
    depends_on('cmake@3.5.2', type='build')
```
```suggestion
    depends_on('cmake@3.5:', type='build')
```
Any version newer than 3.5 should work.

File: var/spack/repos/builtin/packages/rocalution/package.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    for ver in ['3.5.0']:
+        depends_on('hip@' + ver,  when='@' + ver)
+        depends_on('rocblas@' + ver, type='build', when='@' + ver)
+        depends_on('rocsparse@' + ver, type='build', when='@' + ver)
+        depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
+        depends_on('comgr@' + ver, type='build', when='@' + ver)
+        depends_on('rocprim@' + ver, type='build', when='@' + ver)
+
+    patch('0001-fix-hip-build-error.patch', when='@3.5.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocalution/package.py]:
```suggestion
    patch('0001-fix-hip-build-error.patch')
```

**Comment by srekolam** [comment]
@adamjstewart , i tried to use depends_on('cmake@3.5:', type='build')                         I see the build failue 'NOTFOUND' error .                                                                                                                       Building CXX object src/CMakeFiles/rocalution.dir/base/host/host_matrix_mcsr.cpp.o
cd /tmp/root/spack-stage/spack-stage-rocalution-3.5.0-bv4mpho4fpqphsw3c6fbdpxfscjslzbj/spack-build/src && /spack/opt/spack/linux-centos7-skylake_avx512/gcc-7.3.1/hip-3.5.0-3gpxb5mp3xscv2hdrd3w3prli2vtg5lt/bin/hipcc -DSUPPORT_HIP -Drocalution_EXPORTS -I/tmp/root/spack-stage/spack-stage-rocalution-3.5.0-bv4mpho4fpqphsw3c6fbdpxfscjslzbj/spack-build/include -O2 -g -DNDEBUG -fPIC NOTFOUND -std=c++11 -o CMakeFiles/rocalution.dir/base/host/host_matrix_mcsr.cpp.o -c /tmp/root/spack-stage/spack-stage-rocalution-3.5.0-bv4mpho4fpqphsw3c6fbdpxfscjslzbj/spack-src/src/base/host/host_matrix_mcsr.cpp
clang-11: error: no such file or directory: 'NOTFOUND'

**Comment by adamjstewart** [comment]
Hmm, that's odd. Is this a known bug in CMake @chuckatkins?

**Comment by haampie** [comment]
Can you run rebase on spack's develop branch? (`git fetch [spack repo]`, `git rebase [spack repo]/develop` and `git push [your repo] develop -f`?) This PR shows over 56 commits, but effectively it's just 1 or 2 commits, right?

**Comment by srekolam** [comment]
@adamjstewart , i will try to submit a new pull request for the above 3 files..it is showing 67 commit etc...so, i will try to clean up and resubmit again..

**Comment by adamjstewart** [comment]
@srekolam the problem is that you are submitting a PR based off of your `develop` branch, which includes every other commit you've made previously. I like to keep my develop branch pristine and then create new branches like `packages/rocalution` off of develop on which I can commit and submit PRs.

**Pull request closed without merging**